config/environments/production.rb

```diff
@@ -24,7 +24,15 @@
   cache_config_content = [
     "REPOSITORY_CLIENT_CACHE_DIRECTORY=#{config.repo_cache_dir}",
     'REPOSITORY_CACHE_EXPIRY_MINUTES=20'
-  ].join("\n")
+  ]
+  if defined?(Registry::Engine)
+    config.registry_cache_dir = '/run/rmt/cache/registry'
+    cache_config_content.append[
+      "REGISTRY_CLIENT_CACHE_DIRECTORY=#{config.registry_cache_dir}",
+      "REGISTRY_CACHE_EXPIRY_MINUTES=#{Settings[:registry].try(:token_expiration) || 480}" # 480: 8 hours in minutes
+    ]
+  end
```

Comment:
This is a nice idea but it will break https://github.com/SUSE/rmt/pull/1139. IN the case where a 3rd party does not set up the registry they still need the scrubber and the scrubber depends on the variables set or the systemd service will exit with an error. While the scrubbing command for he repository path will still work errors will be emitted for `find -mmin -type f -delete` because the keys `REGISTRY_CLIENT_CACHE_DIRECTORY` and `REGISTRY_CACHE_EXPIRY_MINUTES` will not be define the shell will set these to ' ' empty. Defining the variable even the case where there is no registry engine does no harm and as such they should be set always when the Public Cloud path is installed.

Reply:
good catch @rjschwei